doc/client_lua_api.txt

```diff
@@ -1152,6 +1277,114 @@ Can be obtained via `minetest.get_meta(pos)`.
+  Draws a 2D texture.
+    * `rect`: The rectangle to draw the texture in. The texture will be stretched to fit.
+    * `texture`: Either the name of the texture with optional texture modifiers or the
+      `Texture` object to draw. Must be readable.
+    * `clip_rect` (optional): If specified, any part of the texture that extends out of this
+      rectangle will not be drawn.
+    * `source_rect` (optional): Defines a rectangular sub-texture of the texture which will be
+      drawn instead of the entire texture. If nil, the whole texture is drawn. This rectangle
+      is relative to the rectangle of the texture being drawn to.
+    * `middle_rect` (optional): Defines the 9-slice rectangle for the texture. If nil, the
+      texture will be drawn normally. This rectangle is relative to `source_rect`.
+      See https://en.wikipedia.org/wiki/9-slice_scaling for more information.
+    * `recolor` (optional): The color to recolor the texture to when drawing. This uses
+      hardware coloring, so it uses color multiplication and is fast.
+    * `no_alpha` (optional): Whether alpha should not be used when drawing the texture.
+      Default false.
+    * The texture being drawn to must be writable.
+* `draw_text(pos, text[, clip_rect][, style])`: Draws text.
+    * `pos`: The position of the top left of the text.
+    * `text`: The text to draw. Newlines and '\0' are converted to spaces. Color escape codes
+      are stripped.
+    * `clip_rect` (optional): If specified, any part of the text that extends out of this
+      rectangle will not be drawn.
+    * `style` (optional): A `FontSpec`.
+    * The texture being drawn to must be writable.
+* `draw_item(rect, item[, clip_rect][, angle])`: Draws an item(stack)
+    * `rect`: The rectangle to fit the item inside. Note that items with certain angles that
+      extend past this rectangle will be clipped to it.
+    * `item`: The itemstring of the item to draw. The item count will not be drawn.
+    * `clip_rect` (optional): If specified, any part of the item that extends out of this
+      rectangle will not be drawn.
+    * `angle` (optional): Sets the 3D angle of the item along each axis.
+    * The texture being drawn to must be writable.
+
+`FontSpec`
+----------
+
+A table describing a font.
+
+### Keys
+* `color`: The color of the text. Default white.
+* `font`: What font to use. Can be "normal" or "mono". Default "normal". If the fallback font
+  is used for the current locale, this option will be overridden.
+* `size`: What size the font should be. The default is user-set.
+	* **Note**: It is recommended to make the size relative to the user-set font size which
+	  can be gotten with `minetest.get_font_size`.
+* `bold`: Whether the text should be bold. Default false.
+* `italic`: Whether the text should be italicized. Default false.
+* `underline`: Whether the text should be underlined. Default false.
+* `strikethrough`: Whether the text should be crossed out. Default false.
+
```

Comment:
`highlight` please?

Reply:
I considered it, but concluded that it's not especially necessary.  It's easy enough to add highlight by using a `minetest.get_text_size` and then `draw_rect` the highlight before drawing the text.  On the other hand, italics, underline, and suchlike can't be done from Lua (underline requires knowledge of the font ascenders), hence their inclusion here.

So, `highlight` can be trivially added if wanted (and it certainly will be in the UI even if not in the drawing API), but it doesn't add any functionality that can't already be done easily enough.  The same goes for why there isn't align/valign functionality built in.

Reply:
> It's easy enough to add highlight by using a minetest.get_text_size and then draw_rect the highlight before drawing the text

You have to manually find the position of the substring, that's painful. Not sure everyone can perfectly place their rects under the substrings. Technically, you can also draw a rect for `underline` and `strikethrough`.

Reply:
`underline` and `strikethrough` require the usage of `CGUITTFont::getAscender`, so they actually can't be done as well Lua-side, not without exposing `getAscender`.  That's why they're included.